thread_t is already defined by OS X

Compiling native platform on OS X (10.11.3) results in the following error:

```
~/src/RIOT/core/include/sched.h:98:24: error: typedef redefinition with different types
      ('struct _thread' vs 'mach_port_t' (aka 'unsigned int'))
typedef struct _thread thread_t;
                       ^
/usr/include/mach/mach_types.h:119:22: note: previous definition is here
typedef mach_port_t             thread_t;
```

#4578 renames `tcb_t` to `thread_t`.

Dependency graph:

`cpu/native/periph/timer.c → mach/mach_host.h → mach/mach_types.h`

Possible fixes:

- rename to `tcb_t`
- rename to `riot_thread_t`
- rename to `thrd_t`
- rename to `process_t`
- `#define thread_t riot_thread_t` in sched.h
- other?

--- a/cpu/native/periph/timer.c
+++ b/cpu/native/periph/timer.c
@@ -27,6 +27,8 @@
 #include <mach/mach_init.h>
 #include <mach/mach_port.h>
 #include <mach/mach_host.h>
+/* Both OS X and RIOT typedef thread_t. timer.c does not use either thread_t. */
+#define thread_t riot_thread_t
 #endif
 
 #include <time.h>

